text/3729-sized-hierarchy.md

```diff
@@ -0,0 +1,1827 @@
+      justified by support for a relatively niche feature like extern types.
+- [rfcs#2255: More implicit bounds (?Sized, ?DynSized, ?Move)][issue_more_implicit_bounds], [kennytm][author_kennytm], Dec 2017
+    - Issue created following [rust#46108][pr_dynsized_rebase] to discuss the
+      complexities surrounding adding new traits which would benefit from relaxed
+      bounds (`?Trait` syntax).
+    - There have been various attempts to introduce new auto traits with
+      implicit bounds, such as `DynSized`, `Move`, `Leak`, etc. Often rejected due to
+      the ergonomic cost of relaxed bounds.
+        - `?Trait` being a negative feature can be confusing to users.
+        - Depending on the specific trait being added as a default bound, downstream crates
+          need to re-evaluate every API to determine if adding `?Trait` makes sense, for
+          each `?Trait` added.
+        - `?Trait` isn't actually fully backwards compatible due to interactions with
+          associated types.
+    - This thread was largely motivated by the `Move` trait and that was
+      replaced by the `Pin` type, but there was an emerging consensus that `DynSized`
+      may be more feasible due to its relationship with `Sized`.
+- [Pre-eRFC: Let's fix DSTs][pre_erfc_fix_dsts], [mikeyhew][author_mikeyhew], Jan 2018
+    - This eRFC was written as a successor to [rfcs#1524][rfc_custom_dst].
+    - It proposes `DynSized` trait and a bunch of others. `DynSized` is a
+      supertrait of `Sized` (indirectly) and contains a `size_of_val` method. This
+      proposal is the first to remove `Sized` bounds if another sized trait (e.g.
+      `DynSized`) has an explicit bound.
+        - This enables deprecation of `?Sized` like this RFC proposes.
+    - A `Thin` type to allow thin pointers to DSTs is also proposed in
+      this pre-eRFC - it is a different `Thin` from the currently unstable
+      `core::ptr::Thin` and it's out-of-scope for this RFC to include a similar type
+      and accepted [rfcs#2580][rfc_pointer_metadata_vtable] overlaps.
+    - This pre-eRFC may be the origin of the idea for a family of `Sized`
+      traits, later cited in [Sized, DynSized, and Unsized][blog_dynsized_unsized].
+    - [rfcs#2510][rfc_pointer_metadata_vtable] was later submitted which was a
+      subset of this proposal (but none of the `DynSized` parts).
+    - This eRFC ultimately fizzled out and didn't seem to result in a proper RFC
+      being submitted.
+- [rfcs#2310: DynSized without ?DynSized][rfc_dynsized_without_dynsized], [kennytm][author_kennytm], Jan 2018
+    - This RFC proposed an alternative version of `DynSized` from
+      [rfcs#1993][rfc_opaque_data_structs]/[rust#44469][pr_dynsized] but without
+      being an implicit bound and being able to be a relaxed bound (i.e. no
+      `?DynSized`).
+        - Adding new implicit bounds which can be relaxed has backwards
+          compatibility hazards, see [rfcs#2255][issue_more_implicit_bounds].
```

Comment:
Again I don't understand what those hazards are supposed to be. Adding bounds that *can* be relaxed is entirely fine. Adding bounds that are automatically relaxed for associated types is a problem, but that's a different proposal.

Anyway this point seems redundant here since #2255 is already discussed separately.
```suggestion
```

Reply:
Fixed